packages/twenty-server/src/engine/integrations/exception-handler/hooks/use-exception-handler.hook.ts

```diff
@@ -125,5 +130,24 @@ export const useExceptionHandler = <PluginContext extends GraphQLContext>(
         },
       };
     },
+    onValidate: ({ context, validateFn, params: { documentAST, schema } }) => {
+      const errors = validateFn(schema, documentAST);
+
+      if (Array.isArray(errors) && errors.length > 0) {
+        // const {workspace} = await options.createContextFactory?.create(context);
+
+        const requestSchemaVersion = context.req.headers['x-schema-version'];
+        const currentSchemaVersion = context.workspace?.currentCacheVersion;
+
+        if (
+          requestSchemaVersion &&
+          requestSchemaVersion !== currentSchemaVersion
+        ) {
+          throw new GraphQLError(
+            `Schema version mismatch: ${requestSchemaVersion}  ${currentSchemaVersion}`,
+          );
```

Comment:
@magrinj i think i might be close 😊 once there is an error, it ends up here, but the schema from `context.workspace` is undefined. The missing piece seems to be the `CreateContextFactory` but im not sure what/where would be the best place to call it. Any ideas? 🤔

Reply:
@rostaklein If needed you should be able to do the same thing in the function `handleResult` above, as you also can access the header of the request and the context 🙂

Reply:
This does not work :( the `handleResult` method does not get called in case the schema is invalid.